src/interfaces.ts

```diff
@@ -28,36 +28,81 @@ export interface SpanielObserverInit {
   USE_NATIVE_IO?: boolean;
 }
 
+export interface TimeCompat {
+  highResTime: number;
+  unixTime: number;
+}
+
 export interface SpanielRecord {
   target: SpanielTrackedElement;
   payload: any;
   thresholdStates: SpanielThresholdState[];
-  lastSeenEntry: IntersectionObserverEntry | null;
+  lastSeenEntry: MaybeInternalIntersectionObserverEntry | null;
 }
 
 export interface SpanielThresholdState {
   lastSatisfied: Boolean;
-  lastEntry: IntersectionObserverEntry | null;
+  lastEntry: MaybeInternalIntersectionObserverEntry | null;
   threshold: SpanielThreshold;
-  lastVisible: number;
+  lastVisible: TimeCompat;
   visible: boolean;
   timeoutId?: number;
 }
 
 export interface SpanielIntersectionObserverEntryInit {
-  time: DOMHighResTimeStamp;
-  rootBounds: ClientRect;
-  boundingClientRect: ClientRect;
-  intersectionRect: ClientRect;
+  highResTime: DOMHighResTimeStamp;
+  unixTime: number;
+  rootBounds: SpanielRect;
+  boundingClientRect: SpanielRect;
+  intersectionRect: SpanielRect & ClientRect;
   target: SpanielTrackedElement;
 }
 
-export interface SpanielObserverEntry extends IntersectionObserverEntryInit {
+export interface SpanielRect extends Partial<DOMRectReadOnly> {
+  readonly height: number;
+  readonly width: number;
+  readonly x: number;
+  readonly y: number;
+}
+
+export interface SpanielObserverEntry {
+  isIntersecting: boolean;
   duration: number;
+  visibleTime: number;
   intersectionRatio: number;
   entering: boolean;
   label?: string;
   payload?: any;
+  unixTime: number;
+  highResTime: number;
+  time: number;
+  target: Element;
+  boundingClientRect: SpanielRect;
+  intersectionRect: SpanielRect;
+  rootBounds: SpanielRect | null;
+}
+
+export interface InternalIntersectionObserverEntry {
+  time: number;
+  highResTime: DOMHighResTimeStamp;
+  target: Element;
+  boundingClientRect: SpanielRect;
+  intersectionRect: SpanielRect;
+  rootBounds: SpanielRect | null;
+  intersectionRatio: number;
+  isIntersecting: boolean;
+}
+
+export interface MaybeInternalIntersectionObserverEntry {
+  time: number;
+  unixTime?: number;
```

Comment:
Can this be removed?

Reply:
I removed `MaybeInternalIntersectionObserverEntry` entirely